src/connector/src/source/cdc/external/postgres.rs

```diff
@@ -385,29 +362,49 @@ impl PostgresExternalTableReader {
         table_name: SchemaTableName,
         start_pk_row: Option<OwnedRow>,
         primary_keys: Vec<String>,
-        limit: u32,
+        scan_limit: u32,
     ) {
         let order_key = Self::get_order_key(&primary_keys);
         let client = self.client.lock().await;
         client.execute("set time zone '+00:00'", &[]).await?;
 
         let stream = match start_pk_row {
             Some(ref pk_row) => {
+                // prepare the scan statement, since we may need to convert the RW data type to postgres data type
+                // e.g. varchar to uuid
+                let prepared_scan_stmt = {
+                    let primary_keys = self
+                        .pk_indices
+                        .iter()
+                        .map(|i| self.rw_schema.fields[*i].name.clone())
+                        .collect_vec();
+
+                    let order_key = Self::get_order_key(&primary_keys);
+                    let scan_sql = format!(
+                        "SELECT {} FROM {} WHERE {} ORDER BY {} LIMIT {scan_limit}",
+                        self.field_names,
+                        Self::get_normalized_table_name(&table_name),
+                        Self::filter_expression(&primary_keys),
+                        order_key,
+                    );
+                    client.prepare(&scan_sql).await?
```

Comment:
Every time it prepares once and execute once? Then let's just use simple statements such as Line 406~413

Reply:
prepare for each `scan_limit` chunk, we need the data type from upstream to convert `varchar` for `uuid`.

Reply:
Can we make it prepare once but only when backfill not finished yet?

Reply:
It is possible,  but needs to refactor the code for example to call a `init` method before doing the snapshot.
We have set the default value of `scan_limit` to 1000, the overhead is low.